src/scripts/users/role-manager.js

```diff
@@ -0,0 +1,181 @@
+        const roleId = roleOptionString.split(EMOJI_ROLE_ID_DELIMITER)[1].match(/\d+/g)[0];
+        const role = message.guild.roles.cache.get(roleId);
+        const roleDescription = getRoleDescription(roleOptionString);
+        return {
+            emojiName: reaction,
+            roleId: role ? role.id : undefined,
+            roleName: role ? role.name : undefined,
+            roleDescription: roleDescription
+        }
+    }).filter((roleOption) => {
+        return roleOption.roleId !== undefined && roleOption.roleName !== undefined;
+    });
+}
+
+function getRoleDescription(roleOptionString) {
+    const maybeDescriptionArray = roleOptionString.split(ROLE_ID_DESCRIPTION_DELIMITER);
+    maybeDescriptionArray.shift();
+    return maybeDescriptionArray.join(ROLE_ID_DESCRIPTION_DELIMITER).trim();
+
+}
+
+function createRoleOptionsForMessage(userMessage, roleOptions) {
+    createRoleOptionsMessage(userMessage, roleOptions)
+        .then(() => {
+            userMessage.delete().catch(console.log);
+        }).catch(console.log);
+}
+
+function createRoleOptionsMessage(message, roleOptions) {
+    const menuOptions = createOptions(roleOptions);
+    const menu = createMenu(message.id, menuOptions);
+    const reactionActionRow = new MessageActionRow().addComponent(menu);
+
+    return message.channel.send(null, {
+        embed: createEmbed(message, roleOptions),
+        components: [reactionActionRow]
+    }).catch(console.log);
+}
+
+function createOptions(roleOptions) {
+    return roleOptions.map((roleOption) => {
+        return createMenuOption(
+            roleOption.roleName,
+            roleOption.roleId,
+            roleOption.emojiName,
+            roleOption.roleDescription
+        );
+    })
+}
+
+function createMenuOption(title, value, emoji, description) {
+    return new MessageMenuOption()
+        .setLabel(title)
+        .setValue(value)
+        .setEmoji(emoji)
+        .setDescription(description);
+}
+
+function createMenu(id, options) {
+    return new MessageMenu()
+        .setID(id)
+        .setPlaceholder(ROLE_SELECTION_MESSAGE)
+        .addOptions(options)
+        .setMaxValues(options.length)
+        .setMinValues('0'); // bug in discord-buttons where setting the minimum to 0 with a number doesn't work
+}
+
+function createEmbed(message, roleOptions) {
+    const content = createEmbedContent(message, roleOptions);
+    const contentLinesArray = content.split('\n');
+    const title = contentLinesArray.shift();
+    const description = contentLinesArray.join('\n').trim();
+
+    return {
+        title: title,
+        description: description,
+        footer: {
+            text: "Select the roles you want from the drop-down list below. Leaving any roles in the list deselected will remove you from those roles."
+        }
+    }
+}
+
+function createEmbedContent(userMessage, roleOptions) {
+    let messageContent = userMessage.content.replace(COMMAND, "");
```

Comment:
`const` ?

Reply:
The `replace` a couple of lines down to construct the new string (with role names used instead of role ids) doesn't modify the original string, it creates a new one which is then reassigned to the same variable so that we can continue replacing ids with names